icu4c/source/tools/genccode/genccode.c

```diff
@@ -131,6 +133,7 @@ main(int argc, char* argv[]) {
             "\t-o or --object      write a .obj file instead of .c\n"
             "\t-m or --match-arch file.o  match the architecture (CPU, 32/64 bits) of the specified .o\n"
             "\t                    ELF format defaults to i386. Windows defaults to the native platform.\n"
+            "\t-c or --cpu-arch    Specify a CPU architecture for which to write a .obj file for ClangCL on Windows\n"
```

Comment:
Wouldn't it be easier to use this tool correctly if this commandline flag was `#ifdef`'ed out altogether on platforms where it isn't implemented?

Reply:
For now, I've only `#ifdef`ed out the parts in `genccode.c`, as I didn't want to make `writeObjectCode` signature different depending on which compiler is used on Windows. I just pass it `NULL` if Clang isn't used. If you think I should make the signature `#ifdef`ed too, let me know.

Reply:
I think it would be better to leave it a generic argument, even if it is only currently valid for ClangCL.  It may be needed in other situations.